src/common/state/redis.cc

```diff
@@ -779,6 +903,15 @@ void redis_task_table_test_and_update_callback(redisAsyncContext *c,
   redisReply *reply = (redisReply *) r;
   /* Parse the task from the reply. */
   Task *task = parse_and_construct_task_from_redis_reply(reply);
+  if (task == NULL) {
```

Comment:
What does this change do, it seems like it changes the behavior of the system? Wouldn't it make sense to still call the callback and let the caller decide how to handle this case in the callback?

Also, if this is only possible with actor tasks, then we should assert that the task is an actor task.